### PR TITLE
Switch Day-1 networking format proposal to nmstate

### PR DESCRIPTION
**Pull request by zaneb**
Use nmstate instead of NetworkManager keyfiles as the user interface for
configuring pre-deploy baremetal host networking.

### CODE REVIEW

File: enhancements/network/baremetal-ipi-network-configuration.md [hunk 1]
```diff
@@ -147,27 +147,19 @@ These are not mutually exclusive. If we implement the install-config option, we
...
 
 For example:
 ```yaml
 openshift-master-0:
-  eth0:
-    [connection]
-    id=eth0
-    uuid=18e0cec7-041c-4fb3-957c-a60c80dd9b85
-    type=ethernet
+  - name: eth0
```
Comment (kirankt) [inline enhancements/network/baremetal-ipi-network-configuration.md]: This should actually be identical to the examples in nmstate repo and kubernetes-nmstate CR data, there are 3 top level fields: interfaces, dns-resolver, routes. So just adding the interface data won't cut it.

https://github.com/nmstate/nmstate/tree/base/examples
https://github.com/nmstate/kubernetes-nmstate/tree/main/docs/examples

```yaml
   interfaces:
   - name: br1
      type: linux-bridge
      state: up
      ipv4:
        dhcp: true
        enabled: true
   ...
   dns-resolver:
      config:
        search:
          ...
   routes:
      ...
```

File: enhancements/network/baremetal-ipi-network-configuration.md [hunk 2]
```diff
@@ -178,12 +170,8 @@ platform:
     hosts:
       - name: openshift-master-0
         networkConfig:
-        - interface: eth0
-          data: |
-            [connection]
-            id=eth0
-            uuid=18e0cec7-041c-4fb3-957c-a60c80dd9b85
-            type=ethernet
+          - name: eth0
```
Comment (kirankt) [inline enhancements/network/baremetal-ipi-network-configuration.md]: ditto

**Review by kirankt — CHANGES_REQUESTED** [review]

File: enhancements/network/baremetal-ipi-network-configuration.md (hunk 1 above)
Comment (zaneb) [reply enhancements/network/baremetal-ipi-network-configuration.md]: Thanks, moved everything under 'interfaces:'.

**Comment by cybertron** [comment]
lgtm. Let's see how this version goes over.

**Pull request merged**